packages/amazonq/test/e2e_new/amazonq/tests/mcp.test.ts

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import '../utils/setup'
+import { WebviewView } from 'vscode-extension-tester'
+import { testContext } from '../utils/testContext'
+import {
+    clickMCPAddButton,
+    clickMCPCloseButton,
+    clickMCPRefreshButton,
+    clickToolsButton,
+    configureMCPServer,
+    saveMCPServerConfiguration,
+} from '../helpers/mcpHelper'
+import { closeAllTabs } from '../utils/cleanupUtils'
+
+describe('Amazon Q MCP Functionality', function () {
+    // this timeout is the general timeout for the entire test suite
+    this.timeout(150000)
+    let webviewView: WebviewView
+
+    before(async function () {
+        webviewView = testContext.webviewView
+    })
+
+    after(async function () {
+        await closeAllTabs(webviewView)
+    })
+
+    it('Test Amazon Q MCP Servers and Built-in Tools Access', async () => {
+        await clickToolsButton(webviewView)
```

Comment:
Does it make sense to create a class that wraps this? I see that we pass in the same value to each function, so wondering if it makes sense to organize it as a class where we pass it to the constructor, and call the methods on it.

Reply:
I was curious if you thought of organizing this differently. I see 
```
await clickToolsButton(webviewView)
await clickMCPAddButton(webviewView)
await configureMCPServer(webviewView)
await saveMCPServerConfiguration(webviewView)
await clickMCPCloseButton(webviewView)
```
But I could imagine:
```
const mcpFormManager = new MCP(webviewView)

await mcpFormManager.clickToolsButton()
...

```
Since all your functions basically prefix with MCP and they take the same argument, it could make sense to organize into a class. I think what you have is fine and totally valid as well.